src/games/kalambury/sidebar/Player.js

```diff
@@ -1,25 +1,24 @@
-import React, { useContext } from "react";
+import React from "react";
 import { Icon, Segment, Feed, Transition, List } from "semantic-ui-react";
 import Action from "./Action";
 import { useTranslation } from "react-i18next";
-import { UserContext } from "contexts/UserContext";
-import { useQuery } from "react-query";
+import { useProfiles } from "contexts/UserContext";
 
 const Player = ({
-  name,
+  uid,
   isActive,
   empty,
-  avatar,
   points,
   actions,
   isWinning,
   isCurrentPlayer,
   handleGuessClick,
 }) => {
   const { t } = useTranslation("kalambury");
-  const { getNickName } = useContext(UserContext);
-  const { data: nickname } = useQuery(["user-nickname", name], (key, id) => getNickName(id));
-  if (!name || empty) {
+  const profiles = useProfiles();
+  const { displayName, photoURL } = profiles.get(uid);
```

Comment:
Maybe we could pass those 2 as props, so this component doesn't have to know about how we persist user data?

Reply:
👍